src/SmartComponents/modules/reducers.js

```diff
@@ -22,47 +79,45 @@ function selectedSystems(selectedIds, selectedSystem) {
 }
 
 function compareReducer(state = initialState, action) {
+    const { payload } = action;
+
     switch (action.type) {
         case `${types.FETCH_COMPARE}_FULFILLED`:
             return {
                 ...state,
                 compare: action.payload,
-                selectedSystemIds: action.payload.systems.map(system => system.id)
+                selectedSystemIds: action.payload.systems.map(system => system.id),
+                filteredData: filterCompareData(action.payload, 1, state.perPage, state.stateFilter, state.factFilter),
+                totalFacts: getTotal(action.payload, state.stateFilter, state.factFilter)
```

Comment:
Any way to save the result of `filterCompareData` to a variable and just take the length?